extensions/amp-install-serviceworker/0.1/amp-install-serviceworker.js

```diff
@@ -105,23 +122,193 @@ export class AmpInstallServiceWorker extends AMP.BaseElement {
     iframe.src = this.iframeSrc_;
     this.element.appendChild(iframe);
   }
+
+  /** @private */
+  maybeInstallUrlRewrite_() {
+    // Only run rewrite in single-doc environment.
+    if (!this.getAmpDoc().isSingleDoc()) {
+      return;
+    }
+
+    const ampdoc = this.getAmpDoc();
+    const win = this.win;
+    const winUrl = parseUrl(win.location.href);
+
+    // Cannot install url rewrite on proxy.
+    if (isProxyOrigin(winUrl)) {
+      return;
+    }
+
+    // Read the url-rewrite config.
+    const urlMatch = this.element.getAttribute(
+        'data-no-service-worker-fallback-url-match');
+    let shellUrl = this.element.getAttribute(
+        'data-no-service-worker-fallback-shell-url');
+    if (!urlMatch && !shellUrl) {
+      return;
+    }
+
+    // Check the url-rewrite config is valid.
+    user().assert(urlMatch && shellUrl,
```

Comment:
Is `https` necessary here? I require doc.origin == shell.origin. Thus, a non-https URLs will still work.